Presentation.Web/app/models/it-system-usage/it-system-usage-life-cycle-status.ts

```diff
@@ -0,0 +1,83 @@
+﻿module Kitos.Models.ItSystemUsage {
+    export enum LifeCycleStatusType {
+        Undecided = 0,
+        NotInUse = 1,
+        PhasingIn = 2,
+        Operational = 3,
+        PhasingOut = 4
+    }
+    
+    export class LifeCycleStatus {
+        static readonly Undecided = {
+            Text: Constants.Select2.EmptyField,
+            EnumStringValue: "Undecided"
+        }
+
+        static readonly PhasingIn = {
+            Text: "Under indfasning",
+            EnumStringValue: "PhasingIn"
+        }
+
+        static readonly Operational = {
+            Text: "I drift",
+            EnumStringValue: "Operational"
+        }
+
+        static readonly PhasingOut = {
+            Text: "Under udfasning",
+            EnumStringValue: "PhasingOut"
+        }
+
+        static readonly NotInUse = {
+            Text: "Ikke i drift",
+            EnumStringValue: "NotInUse"
+        }
+    }
+
+    export class LifeCycleStatusOptions {
+        options: Models.ViewModel.Generic.Select2OptionViewModel<LifeCycleStatusType>[];
+
+        mapValueFromEnum(value: LifeCycleStatusType): Models.ViewModel.Generic.Select2OptionViewModel<LifeCycleStatusType> {
+            switch (value) {
+                case LifeCycleStatusType.Undecided:
+                    return { text: LifeCycleStatus.Undecided.Text, id: value, optionalObjectContext: value };
+                case LifeCycleStatusType.PhasingIn:
+                    return { text: LifeCycleStatus.PhasingIn.Text, id: value, optionalObjectContext: value };
+                case LifeCycleStatusType.Operational:
+                    return { text: LifeCycleStatus.Operational.Text, id: value, optionalObjectContext: value };
+                case LifeCycleStatusType.PhasingOut:
+                    return { text: LifeCycleStatus.PhasingOut.Text, id: value, optionalObjectContext: value };
+                case LifeCycleStatusType.NotInUse:
+                    return { text: LifeCycleStatus.NotInUse.Text, id: value, optionalObjectContext: value };
+                default:
+                    return null;
+            }
+        }
+
+        mapValueFromString(value: string): string {
+            switch (value) {
```

Comment:
you must pre-render a dictionary to perform fast lookups in stead of diong.

This will not perform that nicely
Inspiration can be found here

![image](https://user-images.githubusercontent.com/13116889/191238556-10dd2fad-463b-4d22-82e6-41e912d4c7a3.png)

Reply:
another piece of inspiration from something you made yourself:
![image](https://user-images.githubusercontent.com/13116889/191238817-0aaf5e0d-609e-431c-b7f0-534cff8ddcb2.png)